app/src/main/cpp/skyline/services/timesrv/timezone_manager.cpp

```diff
@@ -0,0 +1,130 @@
+            return result::RuleConversionFailed;
+
+        span(locationName).copy_from(pLocationName);
+
+        return {};
+    }
+
+    ResultValue<SteadyClockTimePoint> TimeZoneManager::GetUpdateTime() {
+        std::lock_guard lock(mutex);
+
+        if (!IsInitialized())
+            return result::ClockUninitialized;
+
+        return updateTime;
+    }
+
+    void TimeZoneManager::SetUpdateTime(const SteadyClockTimePoint &pUpdateTime) {
+        std::lock_guard lock(mutex);
+        updateTime = pUpdateTime;
+    }
+
+    ResultValue<int> TimeZoneManager::GetLocationCount() {
+        std::lock_guard lock(mutex);
+
+        if (!IsInitialized())
+            return result::ClockUninitialized;
+
+        return locationCount;
+    }
+
+    void TimeZoneManager::SetLocationCount(int pLocationCount) {
+        std::lock_guard lock(mutex);
+        locationCount = pLocationCount;
+    }
+
+    ResultValue<std::array<u8, 0x10>> TimeZoneManager::GetBinaryVersion() {
+        std::lock_guard lock(mutex);
+
+        if (!IsInitialized())
+            return result::ClockUninitialized;
+
+        return binaryVersion;
+    }
+
+    void TimeZoneManager::SetBinaryVersion(std::array<u8, 0x10> pBinaryVersion) {
+        std::lock_guard lock(mutex);
+        binaryVersion = pBinaryVersion;
+    }
+
+    Result TimeZoneManager::ParseTimeZoneBinary(span<u8> binary, span<u8> ruleOut) {
+        auto ruleObj{tz_tzalloc(binary.data(), binary.size())};
+        if (!ruleObj)
+            return result::RuleConversionFailed;
+
+        memcpy(ruleOut.data(), ruleObj, ruleOut.size_bytes());
+        tz_tzfree(ruleObj);
+        return {};
+    }
+
+    ResultValue<FullCalendarTime> TimeZoneManager::ToCalendarTime(tz_timezone_t pRule, PosixTime posixTime) {
+        struct tm tmp{};
+        auto posixCalendarTime{tz_localtime_rz(pRule, &posixTime, &tmp)};
+        if (!posixCalendarTime)
+            return result::PermissionDenied; // Not the proper error here but *fine*
+
+        FullCalendarTime out{
+            .calendarTime{
+                .year = static_cast<u16>(posixCalendarTime->tm_year),
+                .month = static_cast<u8>(posixCalendarTime->tm_mon + 1),
+                .day = static_cast<u8>(posixCalendarTime->tm_mday),
+                .hour =  static_cast<u8>(posixCalendarTime->tm_hour),
+                .minute = static_cast<u8>(posixCalendarTime->tm_min),
+                .second = static_cast<u8>(posixCalendarTime->tm_sec),
+            },
+            .additionalInfo{
+                .dayOfWeek = static_cast<u32>(posixCalendarTime->tm_wday),
+                .dayOfYear = static_cast<u32>(posixCalendarTime->tm_yday),
+                .dst = static_cast<u32>(posixCalendarTime->tm_isdst),
+                .gmtOffset = static_cast<i32>(posixCalendarTime->tm_gmtoff),
+            },
+        };
+
+        std::string_view(posixCalendarTime->tm_zone).copy(out.additionalInfo.timeZoneName.data(), strlen(posixCalendarTime->tm_zone));
```

Comment:
Assign the string view as a variable since using it in a `copy` here with `strlen` is doing the same operation twice, use `string_view::size` instead